app/src/main/java/com/weatherxm/ui/common/UIModels.kt

```diff
@@ -502,6 +508,15 @@ data class LineChartData(
 
         return dataSets
     }
+
+    fun getEntryValueForTooltip(position: Float): Float {
+        val value = entries.getOrNull(position.toInt())?.y ?: 0F
+        return if (value.isNaN()) {
```

Comment:
you can use also the `.takeIf()` here, something like 

`return entries.getOrNull(position.toInt())?.y?.takeIf { !it.isNaN() } ?: 0F`